lib/chef/provider/package/dpkg.rb

```diff
@@ -126,34 +137,124 @@ def get_current_version
           return nil
         end
 
+        def get_current_version_from(array)
+          array.map do |name|
+            read_current_version_of_package(name)
+          end
+        end
+
         # Runs command via shell_out_with_timeout with magic environment to disable
-        # interactive prompts. Command is run with default localization rather
-        # than forcing locale to "C", so command output may not be stable.
-        def run_noninteractive(command)
-          shell_out_with_timeout!(command, :env => { "DEBIAN_FRONTEND" => "noninteractive" })
+        # interactive prompts.
+        def run_noninteractive(*command)
+          shell_out_with_timeout!(a_to_s(*command), :env => { "DEBIAN_FRONTEND" => "noninteractive" })
+        end
+
+        # Returns true if all sources exist.  Returns false if any do not, or if no
+        # sources were specified.
+        #
+        # @return [Boolean] True if all sources exist
+        def source_files_exist?
```

Comment:
I think the rest of this patch is going to cause them bigger issues when install_package suddenly starts seeing Arrays, and this will be the least of their issues with this whole patch.

And subclassing is inherently not a public API.  That is why people should use composition and use the public API of the dpkg_package resource directly instead of inheriting and overriding.